packages/babel-types/scripts/generators/validators.js

```diff
@@ -73,6 +84,16 @@ import shallowEqual from "../../utils/shallowEqual";
 import type * as t from "../..";
 import deprecationWarning from "../../utils/deprecationWarning";
 \n`;
+  if (BABEL8) {
+    output += `type Opts<Object> = Partial<{
+  [Prop in keyof Object]: Object[Prop] extends t.Node
+    ? t.Node | Object[Prop]
+    : Object[Prop] extends t.Node[]
+    ? t.Node[] | Object[Prop]
+    : Object[Prop];
+}>;
+\n`;
+  }
```

Comment:
I'm marking this as a `discussion` because I'm not sure we should be using this.
correct vs convenient
We already have a similar method, it's not used.
My current preference is that we don't use it.
https://github.com/babel/babel/blob/e54375e46fd783ed084f75fa1103aaf80945e2ef/packages/babel-types/src/validators/is.ts#L7-L11